fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/common/TransformedRecordSerializerTest.java

```diff
@@ -150,50 +154,63 @@ public void noTransformations() {
         logMetrics("metrics with no transformations");
     }
 
-    @Test
-    public void compressWhenSerializing() {
+    static Stream<Arguments> smallRecords() {
+        return Stream.of(
+                Arguments.of(MySimpleRecord.newBuilder().setRecNo(74545L).build(), Tuple.from(74545L)),
+                Arguments.of(MySimpleRecord.newBuilder().setRecNo(1L).build(), Tuple.from(1L))
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("smallRecords")
+    public void compressSmallRecordWhenSerializing(@Nonnull final MySimpleRecord record, @Nonnull final Tuple primaryKey) {
         TransformedRecordSerializer<Message> serializer = TransformedRecordSerializer.newDefaultBuilder()
                 .setCompressWhenSerializing(true)
                 .setCompressionLevel(Deflater.HUFFMAN_ONLY)
                 .setWriteValidationRatio(1.0)
                 .build();
 
-        // There should be no compression actually added for a small record like this
```

Comment:
I think we want to preserve this comment

Reply:
done!